unit-tests/dds/test-watcher.py

```diff
@@ -0,0 +1,50 @@
+# License: Apache 2.0. See LICENSE file in root directory.
+# Copyright(c) 2022 Intel Corporation. All Rights Reserved.
+
+#test:donotrun:!dds
+
+import pyrs_dds_server as client
+from rspy import log, test
+
+
+import dds
+client.debug( True, 'cli' )
+log.nested = 'cli'
+
+
+#############################################################################################
+#
+test.start( "Checking we can detect another participant..." )
+
+devices_added = 0
+def on_device_added( dev ):
+    global devices_added
+    devices_added += 1
+    log.d( 'added', dev )
+    global watcher
+    watcher.foreach_device( lambda dev: log.d( dev ) )
```

Comment:
Won't this log the first device twice?

Reply:
Yeah. I left it in because it actually found a bug for me.